### ISSUE TITLE
APIs should return error details

### ISSUE THREAD

**Issue by tbnguyen1407** [bluenviron/mediamtx#2518]
## Describe the feature

Currently only a status code is returned for invalid input. Actual error can only be seen in logs.

```
## request
PATCH /v3/config/global/patch
{
    "invalid": "field"
}

## response
Status: 400 Bad Request
Body: empty
```

```
2023/10/17 16:51:07 ERR [API] json: unknown field "invalid"
```

HTTP API should return error details in response body, e.g:

```
{
    "error": "json: unknown field: invalid"
}
```

**Comment by aler9** [comment]
this has been implemented with #2583

**Comment by aler9** [comment]
added in v1.2.1
